tests.py

```diff
@@ -1336,6 +1336,61 @@ def test_gt_with_different_type(self):
         with self.assertRaises(TypeError):
             self.r1 > self.d1
 
+    def test_sequential_comparison(self):
+        d = {}
+        d2 = {}
+        rd = RedisDict(namespace="sequential_comparison")
+
+        # Testing for identity
+        assert d is not d2
+        assert d is not rd
+
+        # Testing for equality
+        assert d == d2
+        assert d == rd
+        assert d.items() == d2.items()
+        assert d.items() == rd.items()
+
+        d["foo1"] = "bar1"
+
+        # Testing for inequality after modification in 'd'
+        assert d != d2
+        assert d != rd
+        assert d.items() != d2.items()
+        assert d.items() != rd.items()
+
+        # Modifying 'd2' and 'rd'
+        d2["foo1"] = "bar1"
+        rd["foo1"] = "bar1"
+
+        # Testing for equality
+        assert d == d2
+        assert d == rd
+        assert d.items() == d2.items()
+        assert d.items() == rd.items()
+
+        d.clear()
+        d2.clear()
+        rd.clear()
+
+        rd.update({"a": {}})
+        d.update({"a": {}})
+        d2.update({"a": {}})
+
+        # Testing for nested comparison
+        assert d == d2
+        assert d == rd
+        assert d.items() == d2.items()
+        assert d.items() == rd.items()
+
+        d.clear()
+
+        # Testing for inequality after clear
+        assert d != d2
+        assert d != rd
+        assert d.items() != d2.items()
+        assert d.items() != rd.items()
+
```

Comment:
Great, we are almost there.

With Pytest, you can use assert, but this is unittest from the standard library.
Thus, we need to change assert into self.assertTrue.

Reply:
Sorry for that oversight! fixed now.